### DISCUSSION TITLE
Is there any way to work with socket.io instance in resolver?

### DISCUSSION THREAD

**Discussion by tmelent** [MichalLytek/type-graphql#917] [Q&A] [answered]
Hi. I need to emit a socket.io event and can't understand really, how can I pass a socket.io (completely same) instance from index.ts to resolver?
What confuses me more, that resolvers and schema are defined before socket.io instance is being created. 
Can anyone help me with this?

**Comment by MichalLytek** [comment, marked as answer]
Put the socket instance into `context` function of apollo server. Then you can get the context object with socket by `@Ctx()` decorator in resolver method.
